### ISSUE TITLE
Log OpenTelemetry Trace ID

### ISSUE THREAD

**Issue by pgillich** [goharbor/harbor#18029]
**Is your feature request related to a problem? Please describe.**
In order to correlate log items of harbor clients and OpenTelemerty traces to harbor logs, the OpenTelemetry Trace ID should be logged (if it's set).

**Describe the solution you'd like**
If the Trace ID is sent to the Harbor in HTTP header or the tracing is enabled (and the Trace ID is generated), the Trace ID should be added to the log lines as a new field, for example:
```
2022-12-27T14:31:38Z [ERROR] [/server/v2.0/handler/user.go:221][requestID="9f2cae97-4d6f-4977-9209-43ba9e073019" traceID="0af7651916cd43dd8448eb211c80319c"]: Failed to get user info for ID 0, error: user 0 not found
```
Where the `taceID` is the given Trace ID.

**Describe the main design/architecture of your solution**
The log middleware should be extended by adding the Trace ID to the fields of context logger. All operations, which use the context logger, `log.G(ctx)`, will log the Trace ID (only if it's given) as a log field

There are 3 use cases:
a) Tracing is not enabled, OpenTelemetry HTTP header `traceparent` is missing: Nothing happen, the `traceID` field is disappeared in the log lines
b) Tracing is enabled: Trace ID is created by trace middleware, or OpenTelemetry HTTP header `traceparent` is used (by otel trace lib). The Span added to the context contains the Trace ID, which will be extracted for the log middleware. The `traceID` field is appeared in the log lines
c) Tracing is not enabled: If OpenTelemetry HTTP header `traceparent` is set by Harbor client, Trace ID is extracted from the header. The `traceID` field is appeared in the log lines

See more details about using OpenTelemetry HTTP header `traceparent` without any OpenTelemetry collector in my article: <https://medium.com/faun/multi-hop-tracing-with-opentelemetry-in-golang-792df5feb37c>.

**Describe the development plan you've considered**
See pull request  #18181 .

**Additional context**
Example `curl` command to use OpenTelemetry HTTP header `traceparent`:
```
curl -u test:Test1234 -H 'traceparent: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01' harbor.kind-01.company.com:88/api/v2.0/users/0
```
The above example code prints out below log line:
```
2022-12-27T14:31:38Z [ERROR] [/server/v2.0/handler/user.go:221][requestID="9f2cae97-4d6f-4977-9209-43ba9e073019" traceID="0af7651916cd43dd8448eb211c80319c"]: Failed to get user info for ID 0, error: user 0 not found
```

**Comment by chlins** [comment]
@pgillich Hi, could you share the difference between the request id and trace id, can request id be helpful to trace the request?

**Comment by pgillich** [comment]
> @pgillich Hi, could you share the difference between the request id and trace id, can request id be helpful to trace the request?

Good question. I dig into the history of X-Request-ID HTTP header field and  W3C [OpenTracing spec](https://www.w3.org/TR/trace-context/#trace-context-http-headers-format).  

In short, this requirement is a bad way and it should be withdrawn.

Details:
The X-Request-ID HTTP header field was the industrial standard for propagating distribution info. But it's dummy and does not support the modern requirements. The OpenTracing introduced new HTTP header fields and skipped the X-Request-ID header field. Of course, the X-Request-ID header can be added to the Span Context as an attribute, see a long conversation at [OpenTelemetry Spec issue 2912](https://github.com/open-telemetry/opentelemetry-specification/issues/2912) and in the Harbor source code in `src/server/middleware/requestid/requestid.go`.

If the X-Request-ID header is set, Harbor uses it and if not set, Harbor generates a random UUID for it, so it's possible to force its value by the client. The value of X-Request-ID header is already used at several places of Harbor source code.
My goal is correlating log items and spans together, so I tried to find the solution for it without changing the Harbor source code. Finally, I found it:

The Harbor client should set the mandatory W3C OpenTracing header (Traceparent) and should set the the X-Request-ID header to Trace ID of Traceparent. Let's see an example request:
```
curl -u 'admin:Harbor12345' -H 'X-Request-ID:0af7651916cd43dd8448eb211c80319d' -H 'traceparent: 00-0af7651916cd43dd8448eb211c80319d-b7ad6b7169203332-01' localhost:88/v2/_catalog
```
Since the X-Request-ID header equals to the Trace ID, the Trace ID can be found in the log as requestID field, see:
```
2023-02-26T17:21:28Z [DEBUG] [/server/middleware/security/basic_auth.go:79][requestID="0af7651916cd43dd8448eb211c80319d"]: a basic auth security context generated for request GET /v2/_catalog
```
Log correlation is fulfilled.

Harbor always adds X-Request-ID header to the Span Context, which can be seen in the tag list of Jaeger Span details, see the screenshot:
<img width="960" alt="Harbor_XreqID" src="https://user-images.githubusercontent.com/954891/221434032-c18691bd-2071-4d38-abb1-7740c47c24d0.png">
Trace correlation at trace collector (for example: Jaeger) is fulfilled.

Perhaps adding the [X-Client-Trace-ID](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#config-http-conn-man-headers-x-client-trace-id) information to Trace State will be useful, but it will be another request.
